In [1]:
import pandas as pd
import numpy as np
from pyecharts.charts import Geo
from pyecharts import options as opts
from pyecharts.globals import ChartType, SymbolType,ThemeType
from pyecharts.charts import Bar,Tab,Line,Map,Timeline
from pyecharts.faker import Faker
from pyecharts import options as opts
from pyecharts.charts import Funnel, Page

In [2]:
df = pd.read_csv('gdp.csv',encoding='utf8',index_col=0)
df1 = pd.read_csv('pgdi.csv',encoding='utf8',index_col=0)
df2 = pd.read_csv('cpi.csv',encoding='utf8',index_col=0)
df3 = pd.read_csv('rengdp.csv',encoding='utf8',index_col=0)

### （1）分省GDP水平

In [3]:
df

2017年     2016年     2015年     2014年     2013年
地区                                                   
北京   28014.94  25669.13  23014.59  21330.83  19800.81
天津   18549.19  17885.39  16538.19  15726.93  14442.01
河北   34016.32  32070.45  29806.11  29421.15  28442.95
山西   15528.42  13050.41  12766.49  12761.49  12665.25
内蒙古  16096.21  18128.10  17831.51  17770.19  16916.50
辽宁   23409.24  22246.90  28669.02  28626.58  27213.22
吉林   14944.53  14776.80  14063.13  13803.14  13046.40
黑龙江  15902.68  15386.09  15083.67  15039.38  14454.91
上海   30632.99  28178.65  25123.45  23567.70  21818.15
江苏   85869.76  77388.28  70116.38  65088.32  59753.37
浙江   51768.26  47251.36  42886.49  40173.03  37756.59
安徽   27018.00  24407.62  22005.63  20848.75  19229.34
福建   32182.09  28810.58  25979.82  24055.76  21868.49
江西   20006.31  18499.00  16723.78  15714.63  14410.19
山东   72634.15  68024.49  63002.33  59426.59  55230.32
河南   44552.83  40471.79  37002.16  34938.24  32191.30
湖北   35478.09  32665.38  29550.19  27379.22  24791.83
湖南   33902.96  31551.37  28902.21  27037.32  24621.67
广东   89705.23  80854.91  72812.55  67809.85  62474.79
广西   18523.26  18317.64  16803.12  15672.89  14449.90
海南    4462.54   4053.20   3702.76   3500.72   3177.56
重庆   19424.73  17740.59  15717.27  14262.60  12783.26
四川   36980.22  32934.54  30053.10  28536.66  26392.07
贵州   13540.83  11776.73  10502.56   9266.39   8086.86
云南   16376.34  14788.42  13619.17  12814.59  11832.31
西藏    1310.92   1151.41   1026.39    920.83    815.67
陕西   21898.81  19399.59  18021.86  17689.94  16205.45
甘肃    7459.90   7200.37   6790.32   6836.82   6330.69
青海    2624.83   2572.49   2417.05   2303.32   2122.06
宁夏    3443.56   3168.59   2911.77   2752.10   2577.57
新疆   10881.96   9649.70   9324.80   9273.46   8443.84

In [4]:
list(df.index)

['北京',
 '天津',
 '河北',
 '山西',
 '内蒙古',
 '辽宁',
 '吉林',
 '黑龙江',
 '上海',
 '江苏',
 '浙江',
 '安徽',
 '福建',
 '江西',
 '山东',
 '河南',
 '湖北',
 '湖南',
 '广东',
 '广西',
 '海南',
 '重庆',
 '四川',
 '贵州',
 '云南',
 '西藏',
 '陕西',
 '甘肃',
 '青海',
 '宁夏',
 '新疆']

In [5]:
for i in range(2013, 2018):
    print(list(zip(list(df.index),list(df["{}年".format(i)]))))

[('北京', 19800.81), ('天津', 14442.01), ('河北', 28442.95), ('山西', 12665.25), ('内蒙古', 16916.5), ('辽宁', 27213.22), ('吉林', 13046.4), ('黑龙江', 14454.91), ('上海', 21818.15), ('江苏', 59753.37), ('浙江', 37756.59), ('安徽', 19229.34), ('福建', 21868.49), ('江西', 14410.19), ('山东', 55230.32), ('河南', 32191.3), ('湖北', 24791.83), ('湖南', 24621.67), ('广东', 62474.79), ('广西', 14449.9), ('海南', 3177.56), ('重庆', 12783.26), ('四川', 26392.07), ('贵州', 8086.86), ('云南', 11832.31), ('西藏', 815.67), ('陕西', 16205.45), ('甘肃', 6330.69), ('青海', 2122.06), ('宁夏', 2577.57), ('新疆', 8443.84)]
[('北京', 21330.83), ('天津', 15726.93), ('河北', 29421.15), ('山西', 12761.49), ('内蒙古', 17770.19), ('辽宁', 28626.58), ('吉林', 13803.14), ('黑龙江', 15039.38), ('上海', 23567.7), ('江苏', 65088.32), ('浙江', 40173.03), ('安徽', 20848.75), ('福建', 24055.76), ('江西', 15714.63), ('山东', 59426.59), ('河南', 34938.24), ('湖北', 27379.22), ('湖南', 27037.32), ('广东', 67809.85), ('广西', 15672.89), ('海南', 3500.72), ('重庆', 14262.6), ('四川', 28536.66), ('贵州', 9266.39), ('云南', 12814.59), ('

In [11]:

def timeline_map() -> Timeline:
    tl = Timeline()
    for i in range(2013, 2018):
        map0 = (
            Map()
            .add(
                "分省GDP/亿元",list(zip(list(df.index),list(df["{}年".format(i)]))),"china",is_map_symbol_show = False
            )
            .set_global_opts(
                title_opts=opts.TitleOpts(title="{}dependency ratio".format(i),subtitle="",
                                          subtitle_textstyle_opts=opts.TextStyleOpts(color="red",font_size=18,font_style="italic")),
                visualmap_opts=opts.VisualMapOpts(min_= 800,max_=90000),
            
            )
        )
        tl.add(map0,"{}年".format(i))
    return tl
timeline_map().render_notebook()

### 数据分析
分省年度GDP数据展现了各省GDP数值差距之大，GDP数值最小的西藏从2013年815亿元增长到2017年的1310亿元，数据显示了两个现象，1.西藏自治区的GDP在全国处于低水平。2.西藏自治区GDP在不断的增长中。GDP数值最大的广东省从2013年62474亿元到2017年增长到89705亿元，广东省的GDP也在在不断的增长。
### 数据得出结论：
中国分省GDP差距大，同时中国各省GDP在不断的增长，中国的经济不断向前发展着。

### （2）分省人均GDP水平

In [7]:
df3

2017年   2016年   2015年   2014年   2013年
地区                                         
北京   128994  118198  106497   99995   94648
天津   118944  115053  107960  105231  100105
河北    45387   43062   40255   39984   38909
山西    42060   35532   34919   35070   34984
内蒙古   63764   72064   71101   71046   67836
辽宁    53527   50791   65354   65201   61996
吉林    54838   53868   51086   50160   47428
黑龙江   41916   40432   39462   39226   37697
上海   126634  116562  103796   97370   90993
江苏   107150   96887   87995   81874   75354
浙江    92057   84916   77644   73002   68805
安徽    43401   39561   35997   34425   32001
福建    82677   74707   67966   63472   58145
江西    43424   40400   36724   34674   31930
山东    72807   68733   64168   60879   56885
河南    46674   42575   39123   37072   34211
湖北    60199   55665   50654   47145   42826
湖南    49558   46382   42754   40271   36943
广东    80932   74016   67503   63469   58833
广西    38102   38027   35190   33090   30741
海南    48430   44347   40818   38924   35663
重庆    63442   58502   52321   47850   43223
四川    44651   40003   36775   35128   32617
贵州    37956   33246   29847   26437   23151
云南    34221   31093   28806   27264   25322
西藏    39267   35184   31999   29252   26326
陕西    57266   51015   47626   46929   43117
甘肃    28497   27643   26165   26433   24539
青海    44047   43531   41252   39671   36875
宁夏    50765   47194   43805   41834   39613
新疆    44941   40564   40036   40648   37553

In [8]:
for i in range(2013, 2018):
    print(list(zip(list(df1.index),list(df1["{}年".format(i)]))))

[('北京', 40830.04), ('天津', 26359.2), ('河北', 15189.64), ('山西', 15119.72), ('内蒙古', 18692.89), ('辽宁', 20817.84), ('吉林', 15998.12), ('黑龙江', 15903.45), ('上海', 42173.64), ('江苏', 24775.54), ('浙江', 29774.99), ('安徽', 15154.31), ('福建', 21217.95), ('江西', 15099.68), ('山东', 19008.26), ('河南', 14203.71), ('湖北', 16472.46), ('湖南', 16004.9), ('广东', 23420.75), ('广西', 14082.3), ('海南', 15733.28), ('重庆', 16568.67), ('四川', 14230.99), ('贵州', 11083.06), ('云南', 12577.87), ('西藏', 9740.43), ('陕西', 14371.55), ('甘肃', 10954.4), ('青海', 12947.84), ('宁夏', 14565.78), ('新疆', 13669.62)]
[('北京', 44488.57), ('天津', 28832.29), ('河北', 16647.4), ('山西', 16538.32), ('内蒙古', 20559.34), ('辽宁', 22820.15), ('吉林', 17520.39), ('黑龙江', 17404.39), ('上海', 45965.83), ('江苏', 27172.77), ('浙江', 32657.57), ('安徽', 16795.52), ('福建', 23330.85), ('江西', 16734.17), ('山东', 20864.21), ('河南', 15695.18), ('湖北', 18283.23), ('湖南', 17621.74), ('广东', 25684.96), ('广西', 15557.08), ('海南', 17476.46), ('重庆', 18351.9), ('四川', 15749.01), ('贵州', 12371.06), ('云南', 1377

In [9]:
def funnel_base() -> Funnel:
    tl = Timeline()
    for i in range(2013,2018):
        c = (
        Funnel()
        .add("分省人均GDP/元", list(zip(list(df3.index),list(df3["{}年".format(i)]))))
        .set_global_opts(
                title_opts=opts.TitleOpts(title="{}分省人均GDP/元".format(i),subtitle="",
                                          subtitle_textstyle_opts=opts.TextStyleOpts(color="red",font_size=18,font_style="italic")),
                visualmap_opts=opts.VisualMapOpts(min_= 23000,max_=130000)
            
            )
        )
        tl.add(c,"{}年".format(i))
    return tl

In [12]:
funnel_base().render_notebook()

### 现象分析
+ 历年数据，中国分省人均GDP不断增长。以2017年为例可以看出中国分省人均GDP水平分布不均，最高的上海市可以达到126634元，最低的甘肃仅有28497元。

### 结论：
+ 中国经济的发展，在人均GDP创造上可以体现，同时在经济高速发展的过程中，也造成了不平衡的状态，以甘肃、西藏、云南和贵州等偏远地区享受到的发展速度远远低于东部沿海地区，中国经济发展展现了不平衡的状态。
+ 地区发展的不平衡始终是中国经济发展的缺陷，缩小地区发展差距，提高全民幸福感应该是国家的要务。

### （3）分省居民人均可支配收入

In [11]:
df1

2018年     2017年     2016年     2015年     2014年     2013年
地区                                                             
北京   62361.22  57229.83  52530.38  48457.99  44488.57  40830.04
天津   39506.15  37022.33  34074.46  31291.36  28832.29  26359.20
河北   23445.65  21484.13  19725.42  18118.09  16647.40  15189.64
山西   21990.14  20420.01  19048.88  17853.67  16538.32  15119.72
内蒙古  28375.65  26212.23  24126.64  22310.09  20559.34  18692.89
辽宁   29701.45  27835.44  26039.70  24575.58  22820.15  20817.84
吉林   22798.37  21368.32  19966.99  18683.70  17520.39  15998.12
黑龙江  22725.85  21205.79  19838.50  18592.65  17404.39  15903.45
上海   64182.65  58987.96  54305.35  49867.17  45965.83  42173.64
江苏   38095.79  35024.09  32070.10  29538.85  27172.77  24775.54
浙江   45839.84  42045.69  38529.00  35537.09  32657.57  29774.99
安徽   23983.58  21863.30  19998.10  18362.57  16795.52  15154.31
福建   32643.93  30047.75  27607.93  25404.36  23330.85  21217.95
江西   24079.68  22031.45  20109.56  18437.11  16734.17  15099.68
山东   29204.61  26929.94  24685.27  22703.19  20864.21  19008.26
河南   21963.54  20170.03  18443.08  17124.75  15695.18  14203.71
湖北   25814.54  23757.17  21786.64  20025.56  18283.23  16472.46
湖南   25240.75  23102.71  21114.79  19317.49  17621.74  16004.90
广东   35809.90  33003.29  30295.80  27858.86  25684.96  23420.75
广西   21485.03  19904.76  18305.08  16873.42  15557.08  14082.30
海南   24579.04  22553.24  20653.44  18978.97  17476.46  15733.28
重庆   26385.84  24152.99  22034.14  20110.11  18351.90  16568.67
四川   22460.55  20579.82  18808.26  17220.96  15749.01  14230.99
贵州   18430.18  16703.65  15121.15  13696.61  12371.06  11083.06
云南   20084.19  18348.34  16719.90  15222.57  13772.21  12577.87
西藏   17286.06  15457.30  13639.24  12254.30  10730.22   9740.43
陕西   22528.26  20635.21  18873.74  17394.98  15836.75  14371.55
甘肃   17488.39  16011.00  14670.31  13466.59  12184.71  10954.40
青海   20757.26  19001.02  17301.76  15812.70  14373.98  12947.84
宁夏   22400.42  20561.66  18832.28  17329.09  15906.78  14565.78
新疆   21500.24  19975.10  18354.65  16859.11  15096.62  13669.62

In [12]:

for i in range(2013, 2018):
    print(list(zip(list(df2.index),list(df2["{}年".format(i)]))))

[('北京', 33337), ('天津', 26261), ('河北', 11557), ('山西', 12078), ('内蒙古', 17168), ('辽宁', 20156), ('吉林', 13676), ('黑龙江', 12978), ('上海', 39223), ('江苏', 23585), ('浙江', 24771), ('安徽', 11618), ('福建', 17115), ('江西', 11910), ('山东', 16728), ('河南', 11782), ('湖北', 13912), ('湖南', 12920), ('广东', 23739), ('广西', 11710), ('海南', 11712), ('重庆', 15423), ('四川', 12485), ('贵州', 9541), ('云南', 11224), ('西藏', 6275), ('陕西', 13206), ('甘肃', 9616), ('青海', 12070), ('宁夏', 13537), ('新疆', 11401)]
[('北京', 36057), ('天津', 28492), ('河北', 12171), ('山西', 12622), ('内蒙古', 19827), ('辽宁', 22260), ('吉林', 13663), ('黑龙江', 15215), ('上海', 43007), ('江苏', 28316), ('浙江', 26885), ('安徽', 12944), ('福建', 19099), ('江西', 12000), ('山东', 19184), ('河南', 13078), ('湖北', 15762), ('湖南', 14384), ('广东', 24582), ('广西', 12944), ('海南', 12915), ('重庆', 17262), ('四川', 13755), ('贵州', 11362), ('云南', 12235), ('西藏', 7205), ('陕西', 14812), ('甘肃', 10678), ('青海', 13534), ('宁夏', 15193), ('新疆', 12435)]
[('北京', 39200), ('天津', 32595), ('河北', 12829), ('山西', 14364), ('内蒙古',

In [13]:
def funnel_base() -> Funnel:
    tl = Timeline()
    for i in range(2013,2018):
        c = (
        Funnel()
        .add("人均可支配收入/元", list(zip(list(df1.index),list(df1["{}年".format(i)]))))
        .set_global_opts(
                title_opts=opts.TitleOpts(title="{}人均可支配收入/元".format(i),subtitle="",
                                          subtitle_textstyle_opts=opts.TextStyleOpts(color="red",font_size=18,font_style="italic")),
                visualmap_opts=opts.VisualMapOpts(min_= 9000,max_=65000)
            
            )
        )
        tl.add(c,"{}年".format(i))
    return tl

In [12]:
funnel_base().render()

'C:\\Users\\miku\\Desktop\\IV_final_data\\render.html'

### 现象分析 
+ 各省人均可支配收入都在不断地增长。
+ 分省人均可支配收入差距巨大，最多的地区和最少的地区差不多有四倍多的差距。

### 结论
+ 从中国GDP和分省人均GDP数据 中，我们已经得出了中国经济向前发展的结论。在人均可支配收入数据中，分省人均可支配收入的增长显示了中国的经济发展落实到了人民群众的身上，人民群众享受到了中国经济发展带来的成就。
+ 危机的存在：中国经济在不断的发展，人民群众的收入也在增长是不可否认的事实，但是不容忽视的是人均的差距，在分省人均可支配收入中，足足四倍的差距仿佛一根刺，扎在了中国经济发展的动脉上。中国要实现两个“一百年目标”要保持经济增长的同时，也不可忽视人均差距的存在。

### （4）分省居民消费水平

In [13]:
df2

2017年  2016年  2015年  2014年  2013年
地区                                    
北京   52912  48883  39200  36057  33337
天津   38975  36257  32595  28492  26261
河北   15893  14328  12829  12171  11557
山西   18132  15065  14364  12622  12078
内蒙古  23909  22293  20835  19827  17168
辽宁   24866  23670  23693  22260  20156
吉林   15083  13786  14630  13663  13676
黑龙江  18859  17393  16443  15215  12978
上海   53617  49617  45816  43007  39223
江苏   39796  35875  31682  28316  23585
浙江   33851  30743  28712  26885  24771
安徽   17141  15466  13941  12944  11618
福建   25969  23355  20828  19099  17115
江西   17290  16040  14489  12000  11910
山东   28353  25860  20684  19184  16728
河南   17842  16043  14507  13078  11782
湖北   21642  19391  17429  15762  13912
湖南   19418  17490  16289  14384  12920
广东   30762  28495  26365  24582  23739
广西   16064  15013  13857  12944  11710
海南   20939  18431  17019  12915  11712
重庆   22927  21032  18860  17262  15423
四川   17920  16013  14774  13755  12485
贵州   16349  14666  12876  11362   9541
云南   15831  14534  13401  12235  11224
西藏   10990   9743   8756   7205   6275
陕西   18485  16657  15363  14812  13206
甘肃   14203  13086  11868  10678   9616
青海   18020  16751  15167  13534  12070
宁夏   21058  18570  17210  15193  13537
新疆   16736  15247  13684  12435  11401

In [14]:
def funnel_base() -> Funnel:
    tl = Timeline()
    for i in range(2013,2018):
        c = (
        Funnel()
        .add("居民消费水平/元", list(zip(list(df2.index),list(df2["{}年".format(i)]))))
        .set_global_opts(
                title_opts=opts.TitleOpts(title="{}居民消费水平/元".format(i),subtitle="",
                                          subtitle_textstyle_opts=opts.TextStyleOpts(color="red",font_size=18,font_style="italic")),
                visualmap_opts=opts.VisualMapOpts(min_= 9500,max_=54000)
            
            )
        )
        tl.add(c,"{}年".format(i))
    return tl

In [15]:
funnel_base().render()

'C:\\Users\\miku\\Desktop\\IV_final_data\\render.html'

### 数据分析
+ 各省居民消费水平不断增长。
+ 各省居民消费水平差距巨大，最大差距达5至6倍之大。

### 结论
+ 居民消费水平真正体现了居民享受经济发展的成果水平，结合人均GDP创造数据，可以看出中国经济发展的创造和经济发展成果的享受呈现了同比增长的趋势。
+ 原本经济发展较落后的地区在持续的经济增长态势中，依旧处于落后的趋势，如果不加以干预，落后地区与发达地区的差距会愈加扩大，容易造成社会分层，带来社会不安因素。

In [18]:
list(df.index)

['北京',
 '天津',
 '河北',
 '山西',
 '内蒙古',
 '辽宁',
 '吉林',
 '黑龙江',
 '上海',
 '江苏',
 '浙江',
 '安徽',
 '福建',
 '江西',
 '山东',
 '河南',
 '湖北',
 '湖南',
 '广东',
 '广西',
 '海南',
 '重庆',
 '四川',
 '贵州',
 '云南',
 '西藏',
 '陕西',
 '甘肃',
 '青海',
 '宁夏',
 '新疆']

In [19]:
df.index

Index(['北京', '天津', '河北', '山西', '内蒙古', '辽宁', '吉林', '黑龙江', '上海', '江苏', '浙江',
       '安徽', '福建', '江西', '山东', '河南', '湖北', '湖南', '广东', '广西', '海南', '重庆', '四川',
       '贵州', '云南', '西藏', '陕西', '甘肃', '青海', '宁夏', '新疆'],
      dtype='object', name='地区')

In [20]:
'52912','38975','15893','18132','23909','24866','15083','18859','53617','39796','33851','17141','25969','17290','28353',
'17842','21642','19418','30762','16064','20939','22927','17920','16349','15831','10990','18485','14203','18020','21058','16736',



('17842',
 '21642',
 '19418',
 '30762',
 '16064',
 '20939',
 '22927',
 '17920',
 '16349',
 '15831',
 '10990',
 '18485',
 '14203',
 '18020',
 '21058',
 '16736')

In [21]:
from pyecharts.faker import Faker
from pyecharts import options as opts
from pyecharts.charts import Bar

In [22]:
def bar_base_with_animation() -> Bar:
    c = (
        Bar(
            init_opts=opts.InitOpts(
                animation_opts=opts.AnimationOpts(
                    animation_delay=1000, animation_easing="elasticOut"
                )
            )
        )
        .add_xaxis(['北京', '天津', '河北', '山西', '内蒙古', '辽宁', '吉林', '黑龙江', '上海', '江苏', '浙江',
       '安徽', '福建', '江西', '山东', '河南', '湖北', '湖南', '广东', '广西', '海南', '重庆', '四川',
       '贵州', '云南', '西藏', '陕西', '甘肃', '青海', '宁夏', '新疆'])
        .add_yaxis("2017人均GDP", ['128994','118944','45387','42060','63764','53527','54838','41916','126634','107150','92057','43401','82677','43424','72807',
'46674','60199','49558','80932','38102','48430','63442','44651','37956','34221','39267','57266','28497','44047','50765','44941',])
        .add_yaxis("2017人均消费水平", ['52912','38975','15893','18132','23909','24866','15083','18859','53617','39796','33851','17141','25969','17290','28353',
'17842','21642','19418','30762','16064','20939','22927','17920','16349','15831','10990','18485','14203','18020','21058','16736',])
        .set_global_opts(
            title_opts=opts.TitleOpts(title="人均GDP与人均消费水平", subtitle="/元")
        )
    )
    return c

In [23]:
bar_base_with_animation().render_notebook()

### 数据分析
+ 图抽取了2017年分省人均GDP和2017年分省人均消费水平数据，图中的数据分析显示，人均消费水平一般为人均消费水平的40&的水平。
+ 图中可以分析出人均GDP高的地区，相对的人均消费水平也会相对较高，少的地区也会相对少。

### 结论
+ 中国经济的发展呈现出地区不平衡状态。
+ 中国GDP创造高的地区相对的消费水平也会较高。
+ 中国经济的发展确实落实到了居民的头上，GDP的不断创造之下，居民的消费水平也在不断地提升。
+ 中国经济发展呈现了地区发展不平衡状态，随着经济地不断发展，地区的不平衡状态也在不断地扩大，继续的扩大可能会造成社会性矛盾。

### 建议
+ 坚持经济发展优先原则，继续保持经济的持续增长。
+ 平衡地区发展不平衡状态，扶持中西部经济发展，实现先富带动后富策略，坚持走共同富裕道路。
+ 实行产业转移，将东部过剩的产业转移到中西部，实现产业扶贫，才是可持续化发展的选择。
+ 坚持重点扶贫和精准扶贫策略，实现“两个一百年”奋斗目标。

